query-planner/src/autofrag.rs

```diff
@@ -0,0 +1,305 @@
+use crate::builder::get_field_def_from_type;
+use crate::consts::{MUTATION_TYPE_NAME, QUERY_TYPE_NAME};
+use crate::context::QueryPlanningContext;
+use graphql_parser::query::refs::{
+    FragmentDefinitionRef, FragmentSpreadRef, SelectionRef, SelectionSetRef,
+};
+use graphql_parser::query::{Operation, Selection};
+use graphql_parser::schema::TypeDefinition;
+use graphql_parser::Name;
+use std::collections::hash_map::DefaultHasher;
+use std::collections::HashMap;
+use std::hash::{Hash, Hasher};
```

Comment:
`Hasher` unused?
```suggestion
use std::hash::Hash;
```

Reply:
Importing `Hasher` allows us to use `s.finish()` in line 176.
For future reference, one of the lint checks we do with clippy in the github action will warn about unused imports.